src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include <dwrite_3.h>
+#include <AppInstallerStrings.h>
+#include <winget/Fonts.h>
+#include <winget/Locale.h>
+
+namespace AppInstaller::Fonts
+{
+    namespace
+    {
+        std::vector<std::filesystem::path> GetFontFilePaths(const wil::com_ptr<IDWriteFontFace>& fontFace)
+        {
+            UINT32 fileCount = 0;
+            THROW_IF_FAILED(fontFace->GetFiles(&fileCount, nullptr));
+
+            static_assert(sizeof(wil::com_ptr<IDWriteFontFile>) == sizeof(IDWriteFontFile*));
+            std::vector<wil::com_ptr<IDWriteFontFile>> fontFiles;
+            fontFiles.resize(fileCount);
+
+            THROW_IF_FAILED(fontFace->GetFiles(&fileCount, fontFiles[0].addressof()));
+
+            std::vector<std::filesystem::path> filePaths;
+            for (UINT32 i = 0; i < fileCount; ++i) {
+                wil::com_ptr<IDWriteFontFileLoader> loader;
+                THROW_IF_FAILED(fontFiles[i]->GetLoader(loader.addressof()));
+
+                const void* fontFileReferenceKey;
+                UINT32 fontFileReferenceKeySize;
+                THROW_IF_FAILED(fontFiles[i]->GetReferenceKey(&fontFileReferenceKey, &fontFileReferenceKeySize));
+
+                if (const auto localLoader = loader.try_query<IDWriteLocalFontFileLoader>()) {
+                    UINT32 pathLength;
+                    THROW_IF_FAILED(localLoader->GetFilePathLengthFromKey(fontFileReferenceKey, fontFileReferenceKeySize, &pathLength));
+                    pathLength += 1; // Account for the trailing null terminator during allocation.
+
+                    wchar_t* path = new wchar_t[pathLength];
```

Comment:
Memory leak.  A quick way to know is "using `new` probably isn't the answer".  If you can't solve it with a smart pointer, you are probably holding it wrong.

In this case, you should:
```
std::wstring path;
path.resize(pathLength);
... call the function
path.resize(pathLength - 1); // to remove the null char
```
This way you already have your `std::wstring` and don't need to create it by copying.